src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -5628,6 +5628,91 @@ theorem msb_replicate {n w : Nat} {x : BitVec w} :
   simp only [BitVec.msb, getMsbD_replicate, Nat.zero_mod]
   cases n <;> cases w <;> simp
 
+/-! ### Count leading zeroes -/
+
+theorem clzAux_eq_zero_iff {x : BitVec w} {n : Nat} :
+    clzAux x n = 0 ↔ x.getLsbD n = true := by
```

Comment:
In the next PR, can we write down what these imply for `clz`? It seems useful to build up the basic theory of `clz` in the next PR, since it's an operation we are adding to the BV library.

Reply:
yes, happy to do this in a subsequent PR (once I also upstream all the infra I wrote for `clz`)